Simplenote/src/main/java/com/automattic/simplenote/models/Tag.java

```diff
@@ -79,54 +79,59 @@ public void setIndex(Integer tagIndex) {
 
     public void renameTo(String tagOld, String tagNew, int index, Bucket<Note> notesBucket) throws BucketObjectNameInvalid {
         // When old tag ID is equal to new tag hash, tag is being renamed to lexical variation.
-        boolean isOldIdEqualToNewHash = getSimperiumKey().equals(TagUtils.hashTag(tagNew));
+        boolean isRenamingToLexicalTag = getSimperiumKey().equals(TagUtils.hashTag(tagNew));
```

Comment:
this new name is confusing me. since it may not be obvious, here are how we use the terms:

 - a tag entity exists in the tag bucket, operated on within this `Tag` class
 - tag entities are keyed/referenced by their `entity id`, or `simperiumKey`, which is the `tagHash` - an encoded version of all the possible `lexical variations` of the same tag name
 - the tag `name` is one of multiple possible `lexical variations` of the tag name which result from different ways we can represent a given name in Unicode.

this boolean flag is determining whether or not the new name collapses to the same tag hash as already exists or to a new one. if it's the same tag hash then we can limit our update to the `name` property of the tag. if it collapses to a different hash then we have to take a more complicated approach, creating a new tag with the new hash/name, updating all of the notes which referenced the old tag, and then deleting the old tag.

 - `bookmarks` -> `Bookmarks` is a different lexical variation of the same tag hash `bookmarks` so we can update the `name` property and be done.
 - `bookmarks` -> `favorites` implies updating every note that references `bookmarks` and ensuring that a tag with the `tagHash = "favorites"` exists

this is long-winded but I hope it clarifies

Reply:
Thanks for the detailed reply. It clarifies a lot of things for me.

I'll rename the variable back to `isOldIdEqualToNewHash`. Let me clarify this once again. Please correct me if I'm wrong.

- For a particular tag, `simperiumKey`, `tagHash`, and `entity id` are the same.
- When the hash of the new tag equals to the hash of the old tag, the new tag is either a lexical variation or equal to the old tag.

The reason I thought of renaming that variable is that I saw the following code line in the `TagDialogFragment` class:

https://github.com/Automattic/simplenote-android/blob/edb554cb02993ebf87a7258de92d00c9ab06a07b/Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.java#L101

I'll look into the code once again and will get back to you.